clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -2434,18 +2435,17 @@ void CIRGenModule::buildGlobalDecl(clang::GlobalDecl &D) {
   // ways (e.g. by an extern inline function acquiring a strong function
   // redefinition). Just ignore those cases.
   // TODO: Not sure what to map this to for MLIR
```

Comment:
You can delete the TODO as well. I think this whole change could be further simplified by:
```
auto cirGlobalValue = cast<mlir::cir::CIRGlobalValueInterface>(globalValueOp);
if (!cirGlobalValue.isDeclaration())
  return;
```
See the original code in `CodeGenModule::EmitDeferred()`.

Reply:
Just to see if I understand it correctly. Does this mean "dyn_cast<mlir::cir::GetGlobalOp>(Op)" should not be a valid check here? I.E. early logic in the function should make sure Op is of CIRGlobalValueInterface (either GlobalOp or FuncOp)
Actually, I had that doubt when I changed this code, 
now as you suggested,  after comparing with code in lib/clang/CodeGen/CodeGenModule.cpp
such as 
void CodeGenModule::EmitDeferred() {
and CodeGenModule::GetAddrOfGlobalVar

I kinda question why CIR counterpart CIRGenModule::getAddrOfGlobalVar is returning GetGlobalOp insteadof GlobalOp?

Reply:
My bad, I read `GetGlobalOp` as `GlobalOp`, I take back what I said :) 

> I kinda question why CIR counterpart CIRGenModule::getAddrOfGlobalVar is returning GetGlobalOp insteadof GlobalOp?

The OG one returns `llvm::Constant *`, which if we think in CIR terms, there isn't a good direct counterpart - it could be either and attribute (globalview) or an actual mlir::value result of getglobalop, so not really translatable at the time we wrote this.

If you have better ideas on how to go about this, it's definitely an improvement we could do in its own PR.